library/core/src/slice/index.rs

```diff
@@ -106,6 +105,47 @@ const fn slice_end_index_overflow_fail() -> ! {
     panic!("attempted to index slice up to maximum usize");
 }
 
+// The UbChecks are great for catching bugs in the unsafe methods, but including
+// them in safe indexing is unnecessary and hurts inlining and compile-time.
```

Comment:
The existing perf run doesn't back up a claim that doubling up on the checks hurts compile time. I agree that we shouldn't be checking twice, but I don't think we should be making a claim that's contradicted by the available data.